test/unit/mdc-dialog/foundation.test.js

```diff
@@ -246,17 +249,40 @@ test('on dialog surface click does not close or notify if the event target is no
   td.verify(mockAdapter.notifyAccept(), {times: 0});
 });
 
-test('on click closese the dialog and notifies cancellation', () => {
+test('on click closes the dialog and notifies cancellation if event target is the backdrop', () => {
   const {foundation, mockAdapter} = setupTest();
   const handlers = captureHandlers(mockAdapter, 'registerInteractionHandler');
+  const evt = {
+    stopPropagation: () => {},
+    target: {},
+  };
+
+  td.when(mockAdapter.eventTargetHasClass(evt.target, cssClasses.BACKDROP)).thenReturn(true);
 
   foundation.open();
-  handlers.click();
+  handlers.click(evt);
 
   td.verify(mockAdapter.removeClass(cssClasses.OPEN));
   td.verify(mockAdapter.notifyCancel());
 });
 
+test('on click does not close or notify cancellation if event target is not the backdrop', () => {
```

Comment:
nit: might be worth mentioning for future devs that this effectively means a click on the surface does not close the dialog